core/services/chainlink/application.go

```diff
@@ -242,7 +242,25 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 		registrySyncer.AddLauncher(wfLauncher)
 
 		getLocalNode = wfLauncher.LocalNode
+
+		// We get the underlying implementation of the registry to set the GetLocalNode function.
+		//
+		// The rationale behind this is that the capability registry needs to have serializable methods since it's passed around to standard-capabilities as a GRPC service.
+		if r, ok := opts.CapabilitiesRegistry.(*capabilities.Registry); ok {
```

Comment:
@HenryNguyen5 You can get rid of this error class altogether by making opts.CapabilitiesRegistry accept the concrete registry type. This also happens to be fairly idiomatic Golang (to delay casting something to an interface until as late as possible).

Reply:
Ah okay, that makes sense. I was trying to avoid making changes to other portions of the codebase as much as possible